### PR TITLE
[IOTDB-926] Support reconnection of Session

### PR DESCRIPTION
**Pull request by JackieTien97**

### CODE REVIEW

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 1]
```diff
@@ -239,7 +244,15 @@ public void insertTablet(Tablet tablet, boolean sorted)
     try {
       RpcUtils.verifySuccess(client.insertTablet(request));
     } catch (TException e) {
-      throw new IoTDBConnectionException(e);
+      if (reconnect()) {
```
Comment (neuyilan) [inline session/src/main/java/org/apache/iotdb/session/Session.java:247]: The reconnect() method will call the open() method again, would this cause an infinite loop？
If the open fails, I think we can just throw an exception, as next time the insert request came can call the reconnect() method.

File: session/src/main/java/org/apache/iotdb/session/Config.java [hunk 2]
```diff
@@ -24,5 +24,6 @@
   public static final String DEFAULT_PASSWORD = "password";
   public static final int DEFAULT_FETCH_SIZE = 10000;
   public static final int DEFAULT_TIMEOUT_MS = 0;
-
+  public static final int RETRY_NUM = 3;
+  public static final long RETRY_INTERVAL = 1000;
```
Comment (neuyilan) [inline session/src/main/java/org/apache/iotdb/session/Config.java:28]: RETRY_INTERVAL_MS?

Comment (qiaojialin) [reply session/src/main/java/org/apache/iotdb/session/Config.java:28]: +1

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 1 above)
Comment (JackieTien97) [reply session/src/main/java/org/apache/iotdb/session/Session.java:247]: It seems that the open() doesn't call the reconnect(), there won't be any infinite loops.

Comment (neuyilan) [reply session/src/main/java/org/apache/iotdb/session/Session.java:247]: yeah, I read the code in the wrong place, it indeed does not have the problems.

File: session/src/main/java/org/apache/iotdb/session/Config.java (hunk 2 above)
Comment (JackieTien97) [reply session/src/main/java/org/apache/iotdb/session/Config.java]: fix

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**